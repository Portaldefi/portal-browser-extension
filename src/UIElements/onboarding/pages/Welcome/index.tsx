import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Header, Image, Button } from 'semantic-ui-react';

export default () => {
  const navigate = useNavigate();

  const handleStart = useCallback(() => {
    navigate('/select-action');
  }, []);

  return (
    <>
      <Header className="app-header" as='h1' inverted textAlign='center'>
        Fabric Wallet
        <span style={{fontSize:14, fontWeight:200}}>by </span><Image src="https://portaldefi.com/assets/portal-logo.gif" alt="Portal Logo" />
      </Header>
      <div className='branding'>
        Safe & Stable & ... Wallet
      </div>
      <Button color='teal' size='huge' onClick={handleStart}>Start Wallet</Button>
    </>
  )
}