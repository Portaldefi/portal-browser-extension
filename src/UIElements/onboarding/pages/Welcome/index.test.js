import { render, screen } from '@testing-library/react';
import Welcome from './index';

describe('WelcomeComponent', () => {
  it('should work', async () => {
    let tree;
    act(() => {
      tree = create(
        <Welcome />,
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});