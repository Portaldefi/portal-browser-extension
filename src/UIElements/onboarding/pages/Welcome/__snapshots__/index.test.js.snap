// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeComponent should work 1`] = `
<div
  className="ui segment board"
>
  <div
    className="ui grid"
  >
    <div
      className="centered row"
    >
      <div
        className="ui medium header heading"
      >
        Welcome to Portal
      </div>
    </div>
    <div
      className="centered row"
    >
      <p
        className="ui grey small header description"
      >
        Your Gateway To Uncensorable Applications
      </p>
    </div>
    <div
      className="centered row"
    >
      <img
        className="ui image"
        src="/icons/logo.png"
      />
    </div>
    <div
      className="centered one column row"
    >
      <button
        className="ui button primary-button"
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
</div>
`;
